osu.Game/Skinning/LegacySkin.cs

```diff
@@ -56,6 +57,28 @@ public LegacySkin(SkinInfo skin, IStorageResourceProvider resources)
         protected LegacySkin(SkinInfo skin, IStorageResourceProvider? resources, IResourceStore<byte[]>? fallbackStore, string configurationFilename = @"skin.ini")
             : base(skin, resources, fallbackStore, configurationFilename)
         {
+            if (resources != null
+                && skin.LayoutVersion < 20240625
```

Comment:
How long are these conditionals going to need to live? Are these going to have to live forever or are we going to be able to remove them at some point? If it's the former then I'm even more soured on this entire versioning scheme already. It's basically legacy skin version v2 (let that sink in).

Reply:
https://github.com/ppy/osu/pull/26249#discussion_r1659645108 should be enough to answer this, but at least now it's more isolated inside of `Skin`.